Providers/Profile.swift

```diff
@@ -885,7 +934,14 @@ open class BrowserProfile: Profile {
             case "tabs":
                 // Because clients and tabs share storage, and thus we wipe data for both if we reset either,
                 // we reset the prefs for both at the same time.
-                return TabsSynchronizer.resetClientsAndTabsWithStorage(self.profile.remoteClientsAndTabs, basePrefs: self.prefsForSync)
+                
```

Comment:
For tabs, this will end up calling your new `RustRemoteTabs.resetSync()` method, right? And I guess for clients you will want to call the original code but renamed so it doesn't imply anything about tabs?

Reply:
I just push these changes, but it essence yes. I moved the client related logic from `TabsSynchronizer.resetClientsAndTabsWithStorage` to a new `ClientsSynchronizer.resetClientsAndTabsWithStorage` function which is called after `self.profile.tabs.resetSync` succeeds so that the client data isn't removed until the tabs are reset.